### PR TITLE
Hide implementations of complex data structures in SymtabAPI::Symtab

### PR DESCRIPTION
**Pull request by hainest**
This allows us to more easily modify the parallelism around the big tables (e.g., `indexed_modules`) without bumping the ABI around, meaning we don't have to wait until a major release to make those changes. This also removes internal details like `concurrent.h` which puts TBB into the pubic API and the multiple Boost headers needed for `multi_index`.

Although `ModRangeLookup* mod_lookup();` was a public member, it was never documented, and should never be called by a user. I'm not considering this an API-breaking change.

### CODE REVIEW

**Comment by hainest** [comment]
This will have to wait until https://github.com/dyninst/testsuite/pull/233 is merged.

File: symtabAPI/h/Symtab.h [hunk 1]
```diff
@@ -111,6 +91,9 @@ class SYMTAB_EXPORT Symtab : public LookupInterface,
    friend class relocationEntry;
    friend class Object;
 
+   // Hide implementation details that are complex or add large dependencies
+   std::unique_ptr<symtab_impl> impl;
```
Comment (kupsch) [inline symtabAPI/h/Symtab.h:95]: `const std::unique_ptr...`?  The impl never changes.

**Review by kupsch — APPROVED** [review]
Only suggestion: `impl` could be `const`

File: symtabAPI/h/Symtab.h (hunk 1 above)
Comment (hainest) [reply symtabAPI/h/Symtab.h:95]: Since `Symtab` isn't moveable, that is a good idea.

**Pull request merged**